File: Competitions/UC/Berkeley-2024/contest/contest/torreznos/templates/torreznos_template.cpp
#include <iostream>
#include <vector>
using namespace std;

/**
 * Return the minimum weight of a subset of E that fits the problem statement conditions.
 * N: Number of vertices
 * M: Number of edges
 * F: France
 * S: Spain
 * E: Graph edges in the format [u, v, w].
 */
int solve(int N, int M, int F, int S, vector<vector<int>> E) {
	// TODO: Write your code here.
	return 0;
}

int main() {
	int T;
	cin >> T;
	for (int tc = 0; tc < T; ++tc) {
		int N, M, F, S;
		cin >> N >> M >> F >> S;
		vector<vector<int>> E(M, vector<int>(3));
		for (int i = 0; i < M; ++i)
			for (int j = 0; j < 3; ++j)
				cin >> E[i][j];
		cout << solve(N, M, F, S, E);
	}
	return 0;
}

File: Competitions/UC/Berkeley-2024/contest/contest/torreznos/templates/torreznos_template.py
def solve(N: int, M: int, F: int, S: int, E: list[(int, int, int)]) -> int:
    """
    Return the minimum weight of a subset of E that fits the problem statement conditions.
    N : Number of vertices
    M : Number of edges
    F : France
    S : Spain
    E : Graph edges in the format (u, v, w)
    """
    # YOUR CODE HERE
    return -1


def main():
    T = int(input())
    for _ in range(T):
        N, M, F, S = map(int, input().split())
        E = [tuple(map(int, input().split())) for _ in range(M)]
        print(solve(N, M, F, S, E))


if __name__ == "__main__":
    main()
